engine/src/main/java/org/hibernate/search/query/engine/impl/HSQueryImpl.java

```diff
@@ -670,7 +678,32 @@ else if ( this.sort != null && projection != null ) {
 		);
 	}
 
+	private void validateSortFields(ExtendedSearchIntegrator extendedIntegrator) {
+		SortField[] sortFields = sort.getSort();
+		for ( SortField sortField : sortFields ) {
+			if ( sortField instanceof DistanceSortField ) {
+				validateDistanceSortField( extendedIntegrator, sortField );
+			}
+		}
+	}
 
+	private void validateDistanceSortField(ExtendedSearchIntegrator extendedIntegrator, SortField sortField) {
+		String field = sortField.getField();
+		boolean indexedField = false;
+		for ( Class<?> clazz : classesAndSubclasses ) {
+			IndexedTypeDescriptor indexedTypeDescriptor = extendedIntegrator.getIndexedTypeDescriptor( clazz );
+			FieldDescriptor fieldDescriptor = indexedTypeDescriptor.getIndexedField( field );
+			if ( fieldDescriptor != null ) {
+				indexedField = true;
+				if ( fieldDescriptor.getType() != Type.SPATIAL ) {
+					throw log.distanceSortRequiresSpatialField( field );
+				}
+			}
+		}
+		if ( !indexedField ) {
```

Comment:
Not necessarily.
Take for example `SortTest#testCustomFieldComparatorAscendingSort`, this part in particular:
```Sort sort = new Sort( new SortField( "sum", new SumFieldComparatorSource() ) );```

"sum" is not an indexed field, but everything still works because `SumFieldComparatorSource` does not use the passed field name.

So, at the moment, I don't feel like imposing this validation for all sorts.
Probably, this is something for a separate issue.